src/main/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParser.java

```diff
@@ -307,14 +323,60 @@ public Record parse(final byte[] data) throws ParsingException {
             }
         }
 
+        final ImmutableMap.Builder<String, String> annotationsBuilder = ImmutableMap.builder();
+        annotationsBuilder.putAll(annotations.getOtherAnnotations());
+        annotationsBuilder.put(Key.HOST_DIMENSION_KEY, context.getHost());
+
         return new DefaultRecord.Builder()
                 .setMetrics(variables.build())
                 .setTime(annotations.getEnd())
                 .setId(model.getId())
                 .setAnnotations(mappedAnnotations.build())
+                .setDimensions(extractLegacyDimensions(annotationsBuilder.build()))
                 .build();
     }
 
+    // NOTE: Package private for testing
+    /* package private */com.arpnetworking.metrics.mad.model.Record parseV2gLogLine(final JsonNode jsonNode)
+            throws JsonProcessingException {
+
+        final Version2g model = OBJECT_MAPPER.treeToValue(jsonNode, Version2g.class);
+        final ImmutableMap.Builder<String, Metric> variables = ImmutableMap.builder();
+
+        putVariablesVersion2g(model.getTimers(), MetricType.TIMER, variables);
+        putVariablesVersion2g(model.getCounters(), MetricType.COUNTER, variables);
+        putVariablesVersion2g(model.getGauges(), MetricType.GAUGE, variables);
+
+        return new DefaultRecord.Builder()
+                .setMetrics(variables.build())
+                .setTime(model.getDate())
+                .setId(model.getId())
+                .setAnnotations(ImmutableMap.copyOf(model.getAnnotations()))
+                .setDimensions(ImmutableMap.copyOf(model.getDimensions()))
+                .build();
+    }
+
+    /* package private */static ImmutableMap<String, String> extractLegacyDimensions(final Map<String, String> annotations) {
+
+        final ImmutableMap.Builder<String, String> defaultDimensions = ImmutableMap.builder();
+        for (String dimensionKey : LEGACY_DIMENSION_LIST) {
+            String dimensionValue = annotations.get(dimensionKey);
+            if (null == dimensionValue) {
+                dimensionValue = annotations.get("_" + dimensionKey);
```

Comment:
I don't believe this was supported in all formats. I believe the underscore convention was specific to just one previous format (e?).